packages/frontend/src/utils/format.ts

```diff
@@ -2,24 +2,35 @@ import { BigNumber, FixedNumber, utils } from 'ethers'
 import Network from 'src/models/Network'
 import { commafy, prettifyErrorMessage, toTokenDisplay } from '.'
 
-export function formatError(error: any, network?: Network) {
-  if (!error) {
-    return
+type PossibleError = {
+  data?: {
+    message?: string
   }
+  message?: string
+}
 
-  let errMsg = 'Something went wrong. Please try again.'
+export function formatError(error: unknown, network?: Network): string {
   if (typeof error === 'string') {
-    errMsg = error
-  } else if (error?.data?.message) {
-    errMsg = error.data.message
-  } else if (error?.message) {
-    errMsg = error.message
+    return prettifyErrorMessage(error)
   }
 
   if (Array.isArray(error) && error.length === 1) {
     return formatError(error[0], network)
   }
 
+  if (typeof error !== 'object' || error === null) {
```

Comment:
suggestion: use double equality to also check for `undefined`

```suggestion
  if (error == null) {
    return ''
  }
```